shared-bindings/adcbuffer/BufferedInput.c

```diff
@@ -0,0 +1,180 @@
+ * THE SOFTWARE.
+ */
+
+#include <string.h>
+#include "shared/runtime/context_manager_helpers.h"
+#include "py/binary.h"
+#include "py/mphal.h"
+#include "py/nlr.h"
+#include "py/objproperty.h"
+#include "py/runtime.h"
+#include "shared-bindings/microcontroller/Pin.h"
+#include "shared-bindings/adcbuffer/BufferedInput.h"
+#include "shared-bindings/util.h"
+
+//| class BufferedInput:
+//|     """Input analog voltage level to supplied buffer using DMA Capture"""
+//|
+//|     def __init__(self, pin: microcontroller.Pin, buffer: WriteableBuffer, *, sample_rate: int = 500000) -> None:
+//|         """Use the BufferedInput on the given pin. Fill the given buffer from ADC read values at the supplied
+//|         sample_rate.
+//|
+//|         :param ~microcontroller.Pin pin: the pin to read from
+//|         :param ~circuitpython_typing.WriteableBuffer buffer: buffer: A buffer for samples
+//|         :param ~int sample_rate: rate: The desired playback sample rate
+//|
+//|     Usage::
+//|
+//|       import board
+//|       import adcbuffer
+//|       import array
+//|
+//|       length = 1000
+//|       mybuffer = array.array("H", [0] * length)
+//|       rate = 500000
+//|       adcbuf = adcbuffer.BufferedInput(board.GP26, mybuffer, rate)
+//|       adcbuf.readmultiple()
+//|       adcbuf.deinit()
+//|       for i in range(length):
+//|           print(i, mybuffer[i])
+//|
+//|       (TODO) The reference voltage varies by platform so use
+//|       ``reference_voltage`` to read the configured setting.
+//|       (TODO) Provide mechanism to read CPU Temperature."""
+//|       ...
+//|
+STATIC mp_obj_t adcbuffer_bufferedinput_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
+    enum { ARG_pin, ARG_buffer, ARG_sample_rate };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_pin,    MP_ARG_OBJ | MP_ARG_REQUIRED },
+        { MP_QSTR_buffer, MP_ARG_OBJ | MP_ARG_REQUIRED },
+        { MP_QSTR_sample_rate, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 500000} },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all_kw_array(n_args, n_kw, all_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    // Validate Pin
+    const mcu_pin_obj_t *pin = validate_obj_is_free_pin(args[ARG_pin].u_obj);
+
+    // Buffer Pointer defined and allocated by user
+    mp_buffer_info_t bufinfo;
+    mp_get_buffer_raise(args[ARG_buffer].u_obj, &bufinfo, MP_BUFFER_READ);
+
+    // signed or unsigned, byte per sample
+    bool signed_samples = bufinfo.typecode == 'b' || bufinfo.typecode == 'h';
+    uint8_t bytes_per_sample = 1;
+
+    // Bytes Per Sample
+    if (bufinfo.typecode == 'h' || bufinfo.typecode == 'H') {
+        bytes_per_sample = 2;
+    } else if (bufinfo.typecode != 'b' && bufinfo.typecode != 'B' && bufinfo.typecode != BYTEARRAY_TYPECODE) {
+        mp_raise_ValueError(translate("sample_source buffer must be a bytearray or array of type 'h', 'H', 'b' or 'B'"));
```

Comment:
Just name the argument. You can leave off "sample_source", which is not the argument name.
```suggestion
        mp_raise_ValueError_varg(translate("%q must`` be a bytearray or array of type 'h', 'H', 'b' or 'B'"), MP_QSTR_buffer);
```

Reply:
Changed as suggested.